apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -37,12 +37,395 @@ type DatadogAgentSpec struct {
+	// Default: false
+	// +optional
+	EnableServiceEndpoints *bool `json:"enableServiceEndpoints,omitempty"`
+
+	// AdditionalConfigs allows adding advanced Prometheus check configurations with custom discovery rules.
+	// +optional
+	AdditionalConfigs *string `json:"additionalConfigs,omitempty"`
+}
+
+// DatadogMonitorFeatureConfig contains the Datadog Monitor feature configuration.
+// DatadogMonitor is run by the Datadog Operator.
+type DatadogMonitorFeatureConfig struct {
+	// Enabled enables Datadog Monitors.
+	// Default: false
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+// Generic support structs
+
+// HostPortConfig contains host port configuration.
+type HostPortConfig struct {
+	// Enabled enables host port configuration
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Port takes a port number (0 < x < 65536) to expose on the host. (Most containers do not need this.)
+	// If HostNetwork is enabled, this value must match the ContainerPort.
+	// +optional
+	Port *int32 `json:"hostPort,omitempty"`
+}
+
+// UnixDomainSocketConfig contains the Unix Domain Socket configuration.
+// +k8s:openapi-gen=true
+type UnixDomainSocketConfig struct {
+	// Enabled enables Unix Domain Socket.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Path defines the socket path used when enabled.
+	// +optional
+	Path *string `json:"hostFilepath,omitempty"`
+}
+
+// Endpoint configures an endpoint and its associated Datadog credentials.
+type Endpoint struct {
+	// URL defines the endpoint URL.
+	URL *string `json:"url,omitempty"`
+
+	// Credentials defines the Datadog credentials used by the External Metrics Server to query Datadog.
+	Credentials *DatadogCredentials `json:"credentials,omitempty"`
+}
+
+// CustomConfig provides a place for custom configuration of the Agent or Cluster Agent, corresponding to datadog.yaml or datadog-cluster.yaml.
+// The configuration can be provided in the ConfigData field as raw data, or referenced in a ConfigMap.
+// Note: `ConfigData` and `ConfigMap` cannot be set together.
+// +k8s:openapi-gen=true
+type CustomConfig struct {
+	// ConfigData corresponds to the configuration file content.
+	ConfigData *string `json:"configData,omitempty"`
+
+	// ConfigMap references an existing ConfigMap with the configuration file content.
+	ConfigMap *ConfigMapConfig `json:"configMap,omitempty"`
+}
+
+// ConfigMapConfig contains ConfigMap information used to store a configuration file.
+// Note: `FileKey` and `Items` cannot be set together.
+type ConfigMapConfig struct {
+	// Name is the name of the ConfigMap.
+	Name string `json:"name,omitempty"`
+
+	// FileKey corresponds to the key used in ConfigMap.Data to store the configuration file content.
+	// +optional
+	FileKey string `json:"fileKey,omitempty"`
+
+	// Items maps a ConfigMap data key to a file path mount.
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	Items []corev1.KeyToPath `json:"items,omitempty"`
```

Comment:
I think we can have both at the same time. 

We need to document it. 

I think also that an specific `openapi` comment exist to have it validated by api server.

Or can we remove the usage of `FileKey string` and only use `Items []corev1.KeyToPath`

Reply:
OK i removed `FileKey`